turbokrist-cli/src/main/java/me/apemanzilla/krist/turbokrist/cli/Controller.java

```diff
@@ -60,7 +77,26 @@ public void run() {
 		});
 		status.start();
 	}
-
+	
+	private void setupTempAddress() throws IOException, KristAPIException {
+		if (options.getPrivatekey() == null) {
+			File privatekeyFile = new File("privatekey");
+			
+			if (privatekeyFile.exists()) {
+				options.setPrivatekey(new String(Files.readAllBytes(privatekeyFile.toPath()), StandardCharsets.UTF_8));
+			} else {
+				// generate a 64-char secure random pw
+				options.setPrivatekey(RandomStringUtils.random(64, 0, ALPHANUM.length(),
+					false, false, ALPHANUM.toCharArray(), new SecureRandom()));
+				
+				Files.write(privatekeyFile.toPath(), options.getPrivatekey().getBytes());
+			}
+		}
+		
+		options.setTempAddress(Address.makeV2Address(options.getPrivatekey()));
+		NodeState.getKrist().login(options.getPrivatekey()); // make sure the addy exists
```

Comment:
`login` returns the address, or `null` if it didn't succeed. Should probably check for that.